source/tools/glkernel-cmd/KernelGeneration.cpp

```diff
@@ -0,0 +1,381 @@
+#include "KernelGeneration.h"
+
+#include <iostream>
+
+#include <glkernel/Kernel.h>
+#include <glkernel/noise.h>
+#include <glkernel/sample.h>
+#include <glkernel/scale.h>
+#include <glkernel/sequence.h>
+#include <glkernel/shuffle.h>
+#include <glkernel/sort.h>
+
+#include <cppexpose/json/JSON.h>
+
+namespace
+{
+    template <typename T>
+    void processPoissonSquare(glkernel::tkernel<T> & kernel, unsigned int num_probes)
+    {
+        std::cout << "Poisson square is not implemented for the specified kernel type. Only kernel2 is supported." << std::endl;
```

Comment:
As you already have cppassist as a depdency, I suggest you use the cppassist logging interfaces for outputs:
* `cppassist::critical`
* `cppassist::error`
* `cppassist::warning`
* `cppassist::info`
* `cppassist::debug`